specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2023-01-01-preview/AlertRuleRecommendations.json

```diff
@@ -0,0 +1,243 @@
+        "description": "Retrieve alert rule recommendations for a resource type.",
+        "operationId": "AlertRuleRecommendations_ListByResourceType",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/TargetTypeParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful request for a list of alert rule recommendations",
+            "schema": {
+              "$ref": "#/definitions/AlertRuleRecommendationsResponse"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List alert rule recommendations for virtual machines at subscription level": {
+            "$ref": "./examples/AlertRuleRecommendations_GetBySubscription_VM.json"
+          },
+          "List alert rule recommendations for Monitoring accounts at subscription level": {
+            "$ref": "./examples/AlertRuleRecommendations_GetBySubscription_MAC.json"
+          }
+        }
+      }
+    }
+  },
+  "parameters": {
+    "ResourceUriParameter": {
+      "name": "resourceUri",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The identifier of the resource.",
+      "x-ms-parameter-location": "method",
+      "x-ms-skip-url-encoding": true
+    },
+    "TargetTypeParameter": {
+      "name": "targetType",
+      "in": "query",
+      "required": true,
+      "type": "string",
+      "description": "The recommendations target type.",
+      "x-ms-parameter-location": "client"
+    }
+  },
+  "definitions": {
+    "AlertRuleRecommendationsResponse": {
+      "description": "Describes the format of Alert Rule Recommendations response.",
+      "type": "object",
+      "required": [
+        "requestedTargetType",
+        "displayInformation",
+        "rulesArmTemplate"
+      ],
+      "properties": {
+        "requestedTargetType": {
+          "type": "string",
+          "description": "The requested target resource type."
+        },
+        "displayInformation": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DisplayInformationItem"
+          },
+          "x-ms-identifiers": [],
+          "description": "The list of display information, used by the UI. Each information matches a rule recommendation."
+        },
+        "rulesArmTemplate": {
+          "$ref": "#/definitions/RulesArmTemplate"
+        }
+      }
+    },
```

Comment:
The schema should follow valid ARM Resource format i.e., AlertRuleRecommendationsResponse should be an array containing list of resource type "AlertRuleRecommendation".

If the response would be an object instead of an array of recommendation objects, meaning there are no list of recommendations, but a single recommendation for a given resource id or subscription, then evaluate using ARM Singleton Pattern here: [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-contracts.md#singleton-resources](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-contracts.md#singleton-resources).

Reply:
Thanks for the information. 
Changed to return a list